src/families/polkadot/deviceTransactionConfig.js

```diff
@@ -0,0 +1,211 @@
+// @flow
+
+import type { AccountLike, Account, TransactionStatus } from "../../types";
+import type { Transaction } from "./types";
+import type { DeviceTransactionField } from "../../transaction";
+import { getMainAccount } from "../../account";
+import { getAccountUnit } from "../../account";
+import { formatCurrencyUnit } from "../../currencies";
+import { isStash } from "./logic";
+
+export type ExtraDeviceTransactionField = {
+  type: "polkadot.validators",
+  label: string,
+};
+
+const getSendFields = ({
+  status: { amount },
+  account,
+  recipient,
+}: {
+  account: AccountLike,
+  parentAccount: ?Account,
+  transaction: Transaction,
+  status: TransactionStatus,
+  recipient: string,
+}) => {
+  const fields = [];
+
+  fields.push({
+    type: "text",
+    label: "Balances",
+    value: "Transfer (keep alive)",
+  });
+
+  fields.push({
+    type: "address",
+    label: "Dest",
+    address: recipient,
+  });
+
+  fields.push({
+    type: "text",
+    label: "Value",
+    value: formatCurrencyUnit(getAccountUnit(account), amount, {
+      showCode: false,
+      disableRounding: true,
+      showAllDigits: true,
+    }),
+  });
+
+  return fields;
+};
+
+function getDeviceTransactionConfig({
+  account,
+  parentAccount,
+  transaction,
+  status,
+}: {
+  account: AccountLike,
+  parentAccount: ?Account,
+  transaction: Transaction,
+  status: TransactionStatus,
+}): Array<DeviceTransactionField> {
+  const { mode, recipient, rewardDestination } = transaction;
+  const { amount } = status;
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  let fields = [];
+
+  switch (mode) {
+    case "send":
+      fields = getSendFields({
+        transaction,
+        status,
+        parentAccount,
+        account,
+        recipient,
+      });
+      break;
+
+    case "nominate":
+      fields.push({
+        type: "text",
+        label: "Staking",
+        value: "Nominate",
+      });
+
+      fields.push({
+        type: "polkadot.validators",
```

Comment:
just FYI: we will work forward to the idea that we might no longer will allow such type. it will likely be only "text" in the future so we could reuse it for the bot logic etc.. (context of #734 )
but in any case, it's ok to keep this for now. i just wonder: is there a specific UI on LLD that would requires to not do a list of "text" ?

Reply:
There is a specific component yes on LLD and LLM.
This is an array of addresses, so we will need a way to override UI element for this field if this is Generically interpreted as StringArray.
I think this would be easy though to find a solution, by changing the way TransactionConfirm component works on front-end (use something like "key" instead of type maybe for rendering, with "type" being a default rendering).

For instance, field definition for polkadot validators could be:
```
{ type: "StringArray", key: "validators", "value": [....], "label": "Targets" }
```

Reply:
for now, let's keep just what you had (polkadot.validators) and this is a topic that I need to work on for all coins (because many coins integration did their own type) and see how we want it to be done. (because we could also just go with doing as .push as there are validators, as they all end up with screens on the device at the end)